core/src/main/scala/org/apache/spark/status/AppStatusStore.scala

```diff
@@ -238,8 +239,16 @@ private[spark] class AppStatusStore(
             val diff = idx - currentIdx
```

Comment:
`diff` could be negative here?

Reply:
Hi @gengliangwang I will update the PR. Actually there are more changes required to fix this. Thanks